src/pudl/metadata/classes.py

```diff
@@ -2028,6 +2045,41 @@ def sort_resource_names(resource: Resource):
 
         return sorted(resources, key=sort_resource_names, reverse=False)
 
+    @cached_property
+    def encoders(self) -> dict[SnakeCase, Encoder]:
+        """Compile a mapping of field names to their encoders, if they exist.
+
+        This dictionary will be used many times, so it makes sense to build it once
+        when the Package is instantiated so it can be reused.
+        """
+        encoded_fields = [
+            field
+            for res in self.resources
+            for field in res.schema.fields
+            if field.encoder
+        ]
+        encoders: dict[SnakeCase, Encoder] = {}
+        for field in encoded_fields:
+            if field.name not in encoders:
+                encoders[field.name] = field.encoder
+            else:
+                pd.testing.assert_frame_equal(encoders[field.name].df, field.encoder.df)
+                assert encoders[field.name].code_fixes == field.encoder.code_fixes
+                assert encoders[field.name].ignored_codes == field.encoder.ignored_codes
```

Comment:
this is all just ensuring that if the field exists more than once in `encoded_fields` the encoding components are exactly the same? Can you add a comment in here about that? it took me a second to understand.

Reply:
Yes, that's right. I'll add a comment.